tests/test_pay.py

```diff
@@ -3261,6 +3261,77 @@ def test_mpp_presplit_routehint_conflict(node_factory, bitcoind):
     l1.rpc.pay(inv)
 
 
+def test_delpay_argument_invalid(node_factory, bitcoind):
+    """
+    This test includes all possible combination of input error inside the
+    delpay command.
+    """
+
+    l1, l2 = node_factory.get_nodes(2)
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay()
+
+    # invoice unpaid
+    inv = l1.rpc.invoice(10 ** 5, 'inv', 'inv')
+    payment_hash = inv["payment_hash"]
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash)
+
+    # payment unpaid with wrong status (pending status is a illegal input)
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'pending')
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'invalid_status')
+
+    l2.rpc.connect(l1.info['id'], 'localhost', l1.port)
+    l2.fund_channel(l1, 10 ** 6)
+
+    bitcoind.generate_block(6)
+    sync_blockheight(bitcoind, [l1, l2])
+
+    wait_for(lambda: len(l2.rpc.listchannels()['channels']) == 2)
+
+    l2.rpc.pay(inv['bolt11'])
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'failed')
+
+    with pytest.raises(RpcError):
+        l2.rpc.delpay(payment_hash, 'pending')
+
+    assert len(l2.rpc.listpays()['pays']) == 1
+
+    # test if the node is still ready
+    payments = l2.rpc.delpay(payment_hash, 'complete')
+
+    assert payments['payments'][0]['bolt11'] == inv['bolt11']
+    assert len(payments['payments']) == 1
+    assert len(l2.rpc.listpays()['pays']) == 0
+
+
+def test_delpay_payment_split(node_factory, bitcoind):
+    """
+    Test behavior of delpay with an MPP
+    """
+    MPP_TARGET_SIZE = 10**7  # Taken from libpluin-pay.c
+    amt = 5 * MPP_TARGET_SIZE
+
+    l1, l2, l3 = node_factory.line_graph(
+        3, fundamount=10**8, wait_for_announce=True,
+        opts={'wumbo': None}
```

Comment:
Wumbo is not needed here: `MPP_TARGET_SIZE` is denominated in msats, whereas the `fundamount` is denominated in sats. So this creates a channel that is 4 orders of magnitude larger than the amount being transferred:

 - Channel capacity: 10**8 * 1000 = 10**11
 - Payment amount: 5 * 10**7

Doesn't hurt though :wink:

Reply:
>Doesn't hurt though

Yes, it is, but it is also my logic error, I discovered only now that `fundamount` is denominated in sats. I think that `10**5` is enough